aim/db/migration/alembic_migrations/versions/aceb1ac13668_vmm_policy.py

```diff
@@ -59,12 +61,16 @@ def upgrade():
     new_vmms = []
     new_phys = []
     with session.begin(subtransactions=True):
-        for vmm in session.query(old_vmm_table).all():
-            new_vmms.append(resource.VMMDomain(type=vmm.type, name=vmm.name,
-                                               monitored=True))
-        for phys in session.query(old_phys_table).all():
-            new_phys.append(resource.PhysicalDomain(name=phys.name,
-                                                    monitored=True))
+        try:
+            for vmm in session.query(old_vmm_table).all():
+                new_vmms.append(resource.VMMDomain(type=vmm.type, name=vmm.name,
+                                                   monitored=True))
+            for phys in session.query(old_phys_table).all():
+                new_phys.append(resource.PhysicalDomain(name=phys.name,
+                                                        monitored=True))
+        except DBError:
+            # We can fail if the tables are not yet there
+            pass
 
     op.drop_table('aim_vmm_domains')
```

Comment:
wouldn't this fail too if the table didn't exist? Also, why wouldn't the table exist? It is created by a previous revision (1249face3348)

Reply:
The bug has two aspects:

Problem 1.
As described in the Issue #230, there is a new connection created by aceb1ac13668 before the commit of the earlier transactions started by the previous session. Since these transactions were not committed, the SQL statement creating the table did not commit too. You can refer to the SQL output attached here: https://github.com/noironetworks/aci-integration-module/files/2206403/postgresql.log-201806211158.txt. 

Problem 2. 
Looking at the logic in aceb1ac13668_vmm_policy.py, it is trying to fetch data from the previous schema and populating into the new schema. This logic could work well for existing table. In fact, the above issue (1) does not affect operation since a table from previous deployment already exists. But for a fresh deployment, the old schema tables do not exist and will be created fresh during migration which is not happening due to (1). 

Solution:
The solution in this commit is to ignore failure to read the old tables. If tables do not exist, the queries will result in a DBError exception which we catch and ignore.

Hope this explanation helps.